lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:string_scanner/string_scanner.dart';
+
+/// A token in this context is a sequence of characters
+/// scanned from the original text separated by a space
+/// or new line.
+@sealed
+class Token {
+  /// Normalized form of the text in [span].
+  final String value;
+
+  /// Denotes the token position.
```

Comment:
Maybe we can say something like:
```suggestion
  /// Index of the token in the list of tokens from the tokenized license text.
  ///
  /// When a license is tokenized it becomes a list of tokens.
  /// The [index] is the offset of this token within the list of tokens returned from [tokenize] when
  /// the original license text was tokenized.
  ///
  /// Example:
  /// ```dart
  /// final tokens = tokenize('hello world');
  /// assert(tokens[0].index == 0);
  /// assert(tokens[1].index == 1);
  /// ```
```

Yes, yes, it is probably overkill to have an example here... but it's a good exercise in documenting things precisely.

Reply:
It's okay to omit example right?